phac-nml/irida-next
# Add Select2 to group and project transfer

## What does this PR do and why?
This PR adds the `Select2` component to group and project transfer. The `transfer` stimulus controller has also been removed as its functionality is handled by `Select2`

## Screenshots or screen recordings
![image](https://github.com/user-attachments/assets/e1892c2e-2416-4544-8a8b-24c2054f3319)

## How to set up and validate locally
1. Create a `project` with a unique name.
2. Navigate to that `project`, and click `Settings` then `General`.
3. Under `Transfer project`, transfer the `project` to any other `namespace`.
4. Verify the transfer was successful.
5. Repeat the above with a `group`.

Other tests:
Verify `errors` still occur.
6. For projects: Navigate to an `Outbreak 2022/2023` project, and transfer that project to a `namespace` that already contains the same `project name`. An error should occur stating a project with the same name already exists
7. For groups: Create a group whose name already exists in within a `namespace`. Transfer the group to the `namespace` that already contains a `group` with the same name as the created group. An error should occur stating a group with that name already exists.

Verify no destination namespaces.
8. Create a new user and sign in.
9. Create a new project under the user namespace.
10. Navigate to `Transfer project` and ensure the `Transfer Project` button is disabled, and the input reads that the user does not have access to any destination namespaces.
11. Repeat the above with a group

Verify empty state.
12. Within a project, navigate to `Transfer project` and type something into the input that doesn't match any of your namespaces. Ensure the dropdown only includes `No namespaces are associated with that name or ID`.
13. Repeat the above for a group

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
